doc/user_guide/transformations.rst

```diff
@@ -645,17 +656,40 @@ below does the same thing as the example in the
         ol.apply(schedule.children[0])
         return psy
 
+In the gocean1.0 API (and in the future the lfric (dynamo0.3) API) an
+optional **trans_alg** function may also be supplied. This function
+accepts **PSyIR** (reprenting the algorithm layer) as an argument and
+returns **PSyIR** i.e.: ::
+
+   def trans_alg(psyir):
```

Comment:
1) I am not sure yet. I think my long term preference would be to pass PSyIR, but for that PSyIR to be domain-specific and for the specialised nodes to have additional information attached to them, rather than having a separate `PSy`, `Invokes`, `Invoke`, `Arg` etc hierarchy. As you say, this would break all the scripts but would probably be a much better solution. This is obviously directly relates with what we plan to do with the LFRic generation of the PSy-layer. We could probably already have conversation about whether this approach makes sense if we look at the gocean PSy-layer seeing as it is much simpler and is further down the PSyIR route.

2) I'm not sure we should assume that the script will do everything in place. We could add it as a restriction but the script is not limited to transformations, the user can do what they want.